### PR TITLE
[hotfix] [table] Fix a bound problem for array validation

### PR DESCRIPTION
**Pull request by xccui**
## What is the purpose of the change

This PR aims to fix a bound problem for `validateArray()` in `DescriptorProperties`. Currently, the last element of an array will not be validated.

## Brief change log

Changed the loop bound to `<= maxIndex` (see the length validation logic above the updated line).

## Verifying this change

In `DescriptorPropertiesTest.testArrayInvalidValues()`, made the last element an invalid one.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/descriptors/DescriptorProperties.java [hunk 1]
```diff
@@ -1134,7 +1134,7 @@ public void validateArray(String key, Consumer<String> elementValidation, int mi
 		}
 
 		// validate array elements
-		for (int i = 0; i < maxIndex; i++) {
+		for (int i = 0; i < maxIndex + 1; i++) {
```
[the remark this answers is not in the document]
Comment (xccui) [reply flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/descriptors/DescriptorProperties.java]: Thanks for your suggestion, @sunjincheng121.

I'll change to the "less than and equal to" symbol.

**Comment by sunjincheng121** [comment]
@xccui Thanks for the updated.
LGTM. +1 to merged.

Bests,
Jincheng

**Comment by hequn8128** [comment]
@xccui Thanks for fixing the bug. I found another bound problem in `DescriptorProperties.validateFixedIndexedProperties()`.  Would be great if you can also fix it. 

Best, Hequn

File: flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/descriptors/DescriptorProperties.java [hunk 2]
```diff
@@ -1134,7 +1134,7 @@ public void validateArray(String key, Consumer<String> elementValidation, int mi
 		}
 
 		// validate array elements
-		for (int i = 0; i < maxIndex; i++) {
```
Comment (hequn8128) [inline flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/descriptors/DescriptorProperties.java]: @xccui Thanks for fixing the bug. line 941 in this file also contains the bound problem. Would be great if you can also fix it. 

Best, Hequn

Comment (xccui) [reply flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/descriptors/DescriptorProperties.java]: Well, maybe I should create a JIRA issue for these problems. Will close this PR.
Thanks for looking into this, @hequn8128!

Best, Xingcan

**Pull request closed without merging**